### DISCUSSION TITLE
wxMaxima 21.05.2 with Maxima 5.45.1 working on Android 9 (Samsung Galaxy Tab S4)

### DISCUSSION THREAD

**Discussion by tigran123** [wxMaxima-developers/wxmaxima#1594] [Show and tell]
I compiled wxMaxima 21.05.2 with Maxima 5.45.1 (which was compiled using SBCL 2.1.9) inside a proot'ed Debian 10/armhf inside Termux on Android and it works perfectly, see the screenshot below:

![http://www.bibles.org.uk/screenshots/wxMaxima-on-Android-small.png](http://www.bibles.org.uk/screenshots/wxMaxima-on-Android-small.png)

Here is the link to the full-size image (github doesn't seem to like embedding them): http://www.bibles.org.uk/screenshots/wxMaxima-on-Android.png

**Comment by daute** [comment]
Wow, nice. Did you need any modifications, which should be included into wxMaxima's source code? Or some special compilation instructions, which should be included?

Best regards, Wolfgang

**Comment by tigran123** [comment]
Oh yes, sorry, I should have mentioned those:

1. One has to use Termux from F-Droid only (Google Play's version won't work). I used version 0.117.
2. Then I used Andronix to generate Debian 10 rootfs with XFCE desktop environment (other combinations may or may not work, but XFCE definitely does).

Also, I should perhaps mention that if one installs ecl directly in Termux (not in Debian under Termux), then Maxima compiles against it straightaway and runs fine (but I didn't run complete tests). But inside Debian we can compile Maxima against sbcl. We can probably compile it inside Debian against ecl as well, but I didn't bother as I had Maxima/ECL running directly in Termux without going through proot overhead.

**Comment by tigran123** [comment]
Just a small update: as of now Maxima 5.45.1 is part of Termux's set of packages, so there is no need to compile it manually. One just installs it like any other package by `pkg install maxima`. Also, wxwidgets was added recently to termux-x11, so in theory it should be possible to compile wxMaxima directly in termux-x11 without having to go via proot mechanism to boot Debian installation.
